### PR TITLE
[mono][jit] Adding compare all/any intrinsics.

### PR DESCRIPTION
**Pull request by jandupej**
This adds the following intrinsics: `Equals{All,Any}`, `GreaterThan{All,Any}`, `GreaterThanOrEqual{All,Any}`, `LessThan{All,Any}`, `LessThanOrEqual{All,Any}`. 

This performs the vector compare operation, whose result is either all 1 or all 0 in the element depending on the result. Afterwards a horizontal maximum or minimum byte is found to determine if all (or any) are nonzero. `umaxv/uminv` should be fast on M1 with a latency of 3 and throughput of 4 per clock. (https://dougallj.github.io/applecpu/firestorm-simd.html)

Necessary arm64 codegen macros are also fixed and converted to the parametrized form.

Contributes to https://github.com/dotnet/runtime/issues/80566.

### CODE REVIEW

**Review by vargaz — APPROVED** [review]

**Comment by fanyang-mono** [comment]
`SN_op_Equality ` and `SN_op_Inequality` should also be enabled as part of this PR. I think that they are equivalent to `EqualsAll`

**Comment by jandupej** [comment]
Failures are related.

File: src/mono/mono/mini/mini-arm64.c [hunk 1]
```diff
@@ -3350,6 +3350,25 @@ emit_move_return_value (MonoCompile *cfg, guint8 * code, MonoInst *ins)
 	return code;
 }
 
+static guint8*
+emit_xextract (guint8* code, int mode, int dreg, int sreg1)
+{
+	switch (mode) {
+	case SIMD_EXTR_MAX8:
+		arm_neon_umaxv (code, VREG_FULL, TYPE_I8, FP_TEMP_REG, sreg1);
```
[the remark this answers is not in the document]
Comment (jandupej) [reply src/mono/mono/mini/mini-arm64.c:3358]: Yes, there is a 64-bit variant of this.

File: src/mono/mono/mini/simd-intrinsics.c [hunk 2]
```diff
@@ -1190,18 +1190,18 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 		return NULL;
 #endif
 // FIXME: This limitation could be removed once everything here are supported by mini JIT on arm64
-#ifdef TARGET_ARM64
-	if (!(cfg->compile_aot && cfg->full_aot && !cfg->interp))
-		return NULL;
-#endif
+//#ifdef TARGET_ARM64
```
[the remark this answers is not in the document]
Comment (jandupej) [reply src/mono/mono/mini/simd-intrinsics.c:1199]: I like the idea. This is now tracked in https://github.com/dotnet/runtime/issues/83587

File: src/mono/mono/mini/simd-intrinsics.c [hunk 3]
```diff
@@ -1475,18 +1485,26 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
...
-			case SN_EqualsAll:
-				return emit_xequal (cfg, arg_class, args [0], args [1]);
-			case SN_EqualsAny: {
-				MonoInst *cmp_eq = emit_xcompare (cfg, arg_class, arg0_type, args [0], args [1]);
-				MonoInst *zero = emit_xzero (cfg, arg_class);
-				return emit_not_xequal (cfg, arg_class, cmp_eq, zero);
+		if (id == SN_Equals)
+			return emit_xcompare (cfg, klass, arg0_type, args [0], args [1]);
+
+		if (COMPILE_LLVM (cfg)) {
```
[the remark this answers is not in the document]
Comment (jandupej) [reply src/mono/mono/mini/simd-intrinsics.c:1497]: This is now tracked in https://github.com/dotnet/runtime/issues/83588

**Review by fanyang-mono — APPROVED** [review]

**Pull request merged**